app/main.py

```diff
@@ -1,3 +1,52 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float | int) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        return Distance(km=self.km + other)
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, number: int | float) -> Distance:
+        return Distance(km=self.km * number)
+
+    def __truediv__(self, number: int | float) -> Distance:
+        return Distance(km=round(self.km / number, 2))
+
+    def __lt__(self, other: Distance | float | int) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        return self.km < other
+
+    def __gt__(self, other: Distance | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        return self.km > other
+
+    def __eq__(self, other: Distance | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        return self.km == other
+
+    def __le__(self, other: Distance | float) -> bool:
+        return not Distance.__gt__(self, other)
+
+    def __ge__(self, other: Distance | float) -> bool:
+        return not Distance.__lt__(self, other)
```

Comment:
u are close)
You can call these magic methods using operands `< or > ` 
instead of `Distance.__lt__(self, other) or Distance.__gt__(self, other)`